TSOClient/tso.simantics/NetPlay/EODs/Handlers/VMEODRoulettePlugin.cs

```diff
@@ -0,0 +1,1461 @@
+                        Controller.SendOBJEvent(new VMEODEvent((short)VMEODRouletteEvents.CroupierCollectChips, Croupier.Avatar.ObjectID));
+                        // pay the payouts
+                        SettleAccounts(false);
+                        break;
+                    }
+                case VMEODRouletteGameStates.Spinning:
+                    {
+                        Tock = 0;
+                        GameState = state;
+
+                        // send event for croupier to spin the wheel
+                        Controller.SendOBJEvent(new VMEODEvent((short)VMEODRouletteEvents.CroupierSpinWheel, Croupier.Avatar.ObjectID));
+
+                        int winningNumber = NextBall.Next(0, 38); // not sure if this is acceptable and true to the game, but whatever for now
+                        if (winningNumber == 37)
+                            winningNumber = 100; // double zero
+                        // accept the bets of each player, check if they are a winner, send them spinning event
+                        foreach (var player in Players)
+                        {
+                            int playerBetAmount = player.AcceptBets();
+                            if (playerBetAmount > 0)
+                            {
+                                DeductBetAmount(playerBetAmount, player);
+                                player.IsWinner(winningNumber);
+                                // send event to display bet amount taken over head
+                                Controller.SendOBJEvent(new VMEODEvent((short)VMEODRouletteEvents.PlayerShowBetAmount,
+                                    new short[] { player.Client.Avatar.ObjectID, (short)playerBetAmount }));
+                                // *todo* player.PlayKachingSoundEffect
+                            }
+                            player.Client.Send("roulette_spin", winningNumber + "%" + player.PayoutDue);
+                        }
+                        break;
+                    }
+                default:
+                    {
+                        Tock = 0;
+                        GameState = state;
+                        Roundtimer = 0;
+                        break;
+                    }
+            }
+        }
+        private void UIClosedHandler(string evt, string empty, VMEODClient client)
+        {
+            if (client != null)
+                Server.Disconnect(client);
+        }
+        private void NewGame()
+        {
+            if (TableBalance < MaxBet * 140)
+                CloseTable();
+            //else
+            {
+                foreach (var player in Players)
+                {
+                    if (player.Client != null)
+                    {
+                        if (player.SimoleonBalance >= MinBet)
+                            player.NewGame(new byte[] { (byte)(MinBet / 255), (byte)(MinBet % 255), (byte)(MaxBet / 255), (byte)(MaxBet % 255),
```

Comment:
`(byte)(MinBet >> 8), (byte)(MinBet)`
Use bitshifts for short -> byte array, and a cast to byte does %255 (or &0xFF)